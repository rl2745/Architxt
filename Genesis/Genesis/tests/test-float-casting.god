void init(){}
void update(int f){}

int main() {
  float a;
  float b;
  int c;

  float tmp1;
  int tmp2;

  a = 5.0;
  b = 3;
  c = 2.0;

  print(a + c);
  printfl(a + c);

  tmp1 = a + b;
  printfl(tmp1);

  tmp1 = a + c;
  printfl(tmp1);

  tmp1 = c + c;
  printfl(tmp1);

  tmp2 = a + b;
  print(tmp2);

  tmp2 = a + c;
  print(tmp2);

  tmp2 = c + c;
  print(tmp2);

  return 0;
}
