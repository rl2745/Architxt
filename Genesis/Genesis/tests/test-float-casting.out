7
7.000000
8.000000
7.000000
4.000000
8
7
4
